frame/airdrop/src/lib.rs

```diff
@@ -0,0 +1,977 @@
+				// Fund Airdrop account from creators account
+				T::RecipientFundAsset::transfer(
+					&airdrop.creator,
+					&Self::get_airdrop_account_id(airdrop_id),
+					total_funds.safe_sub(&current_funds)?,
+					false,
+				)?;
+			}
+
+			// Populate `RecipientFunds`
+			recipients.iter().for_each(|(identity, funds, vesting_period, is_funded)| {
+				RecipientFunds::<T>::insert(
+					airdrop_id,
+					identity,
+					RecipientFundOf::<T> {
+						total: *funds,
+						claimed: T::Balance::zero(),
+						vesting_period: *vesting_period,
+						funded_claim: *is_funded,
+					},
+				);
+			});
+
+			TotalAirdropRecipients::<T>::mutate(airdrop_id, |total_airdrop_recipients| {
+				*total_airdrop_recipients = total_recipients;
+			});
+
+			// Update Airdrop statistics
+			let (total_funds, claimed_funds) =
+				Airdrops::<T>::try_mutate(airdrop_id, |airdrop| match airdrop.as_mut() {
+					Some(airdrop) => {
+						airdrop.total_funds = total_funds;
+						airdrop.total_recipients = total_recipients;
+						// Ok(airdrop.total_funds.safe_sub(&airdrop.claimed_funds)?)
+						Ok((airdrop.total_funds, airdrop.claimed_funds))
+					},
+					None => Err(Error::<T>::AirdropDoesNotExist),
+				})?;
+
+			Self::deposit_event(Event::RecipientsAdded {
+				airdrop_id,
+				number: transaction_recipients,
+				unclaimed_funds: total_funds.safe_sub(&claimed_funds)?,
+			});
+
+			Ok(())
+		}
+
+		/// Remove a recipient from an Airdrop.
+		///
+		/// Refunds the creator for the value of the recipient fund.
+		///
+		/// # Errors
+		/// * `AirdropDoesNotExist` - No Airdrop exist that is associated 'airdrop_id'
+		/// * `NotAirdropCreator` - Signer of the origin is not the creator of the Airdrop
+		/// * `RecipientAlreadyClaimed` - The recipient has already began claiming their funds.
+		/// * `RecipientNotFound` - No recipient associated with the `identity` could be found.
+		fn remove_recipient(
+			origin_id: Self::AccountId,
+			airdrop_id: Self::AirdropId,
+			recipient: Self::Recipient,
+		) -> DispatchResult {
+			let airdrop = Self::get_airdrop(&airdrop_id)?;
+			ensure!(airdrop.creator == origin_id, Error::<T>::NotAirdropCreator);
+
+			let airdrop_account = Self::get_airdrop_account_id(airdrop_id);
+			let recipient_fund = Self::get_recipient_fund(airdrop_id, recipient.clone())?;
+
+			ensure!(
+				recipient_fund.claimed == T::Balance::zero(),
+				Error::<T>::RecipientAlreadyClaimed
+			);
+
+			// Update Airdrop details
+			let (creator, total_funds, claimed_funds) =
+				Airdrops::<T>::try_mutate(airdrop_id, |airdrop| match airdrop.as_mut() {
+					Some(airdrop) => {
+						airdrop.total_funds =
+							airdrop.total_funds.saturating_sub(recipient_fund.total);
+						Ok((airdrop.creator.clone(), airdrop.total_funds, airdrop.claimed_funds))
+					},
+					None => Err(Error::<T>::AirdropDoesNotExist),
+				})?;
+
+			TotalAirdropRecipients::<T>::mutate(airdrop_id, |total_airdrop_recipients| {
+				*total_airdrop_recipients -= 1;
+			});
```

Comment:
Check out https://github.com/ComposableFi/composable/blob/main/frame/composable-support/src/abstractions/counter/mod.rs, was recently merged

Reply:
I like this but on the add recipient transaction, we increment by a variable amount